code/modules/mob/living/basic/lavaland/raptor/raptor.dm

```diff
@@ -0,0 +1,330 @@
+	move_resist = MOVE_FORCE_VERY_STRONG
+	faction = list(FACTION_RAPTOR)
+	speak_emote = list("screeches")
+	ai_controller = /datum/ai_controller/basic_controller/raptor
+	///can this mob breed
+	var/can_breed = TRUE
+	///can we ride this mob
+	var/ridable_component = /datum/component/riding/creature/raptor
+	//pet commands when we tame the raptor
+	var/static/list/pet_commands = list(
+		/datum/pet_command/idle,
+		/datum/pet_command/free,
+		/datum/pet_command/point_targeting/attack,
+		/datum/pet_command/follow,
+		/datum/pet_command/point_targeting/fetch,
+	)
+	///things we inherited from our parent
+	var/datum/raptor_inheritance/inherited_stats
+	///our color
+	var/raptor_color
+	///the description that appears in the dex
+	var/dex_description
+	///path of our child
+	var/child_path
+
+/mob/living/basic/mining/raptor/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/obeys_commands, pet_commands)
+	AddElement(\
+		/datum/element/change_force_on_death,\
+		move_resist = MOVE_RESIST_DEFAULT,\
+	)
+	var/static/list/display_emote = list(
+		BB_EMOTE_SAY = list("Chirp chirp chirp!", ),
+		BB_EMOTE_SEE = list("shakes its feathers!", "stretches!"),
+		BB_SPEAK_CHANCE = 5,
+	)
+	ai_controller.set_blackboard_key(BB_BASIC_MOB_SPEAK_LINES, display_emote)
+	if(!mapload)
+		GLOB.raptor_count++
+	inherited_stats = new
+	inherit_properties()
+	RegisterSignal(src, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, PROC_REF(pre_attack))
+	var/static/list/my_food = list(/obj/item/stack/ore)
+	AddElement(/datum/element/basic_eating, food_types = my_food)
+	AddElement(/datum/element/ai_retaliate)
+	AddElement(/datum/element/ai_flee_while_injured, stop_fleeing_at = 0.5, start_fleeing_below = 0.2)
+	if(ridable_component)
+		AddElement(/datum/element/ridable, ridable_component)
+	if(can_breed)
+		AddComponent(\
+			/datum/component/breed,\
+			can_breed_with = typecacheof(list(/mob/living/basic/mining/raptor)),\
+			baby_path = /obj/item/food/egg/raptor_egg,\
+			post_birth = CALLBACK(src, PROC_REF(egg_inherit)),\
+			breed_timer = 3 MINUTES,\
+		)
+	add_happiness_component()
+
+
+/mob/living/basic/mining/raptor/proc/add_happiness_component()
+	var/static/list/percentage_callbacks = list(0, 15, 25, 35, 50, 75, 90, 100)
+	AddComponent(\
+		/datum/component/happiness,\
+		on_petted_change = 50,\
+		on_groom_change = 50,\
+		on_eat_change = 200,\
+		callback_percentages = percentage_callbacks,\
+		happiness_callback = CALLBACK(src, PROC_REF(happiness_change)),\
+	)
+
+/mob/living/basic/mining/raptor/proc/pre_attack(mob/living/puncher, atom/target)
+	SIGNAL_HANDLER
+
+	if(!istype(target, /obj/structure/ore_container/food_trough/raptor_trough))
+		return
+
+	var/obj/ore_food = locate(/obj/item/stack/ore) in target
+
+	if(isnull(ore_food))
+		balloon_alert(src, "no food!")
+	else
+		melee_attack(ore_food)
+	return COMPONENT_HOSTILE_NO_ATTACK
+
+/mob/living/basic/mining/raptor/death(gibbed)
+	. = ..()
+	if(HAS_TRAIT(src, TRAIT_MOB_HATCHED))
+		GLOB.raptor_count--
```

Comment:
ehhh the more i think about it the less i like increment/decrement `raptor_count` variable, imo it should be a list of string `REF()`s pointing towards the mob, we add the mob ref to the list on a non-maploaded spawn and if a raptor dies with their string ref in the list then we remove the entry. then we access `length(GLOB.hatched_raptors)` for the integer value